app/models/form.rb

```diff
@@ -97,6 +97,36 @@ def all_questions
     }.reduce(:merge)
   end
 
+  def filter_conditional_questions(questions, case_log)
```

Comment:
Is this being used in multiple places/ non check_answers page. 
Should the form know about the case_log?

Reply:
It's being used to calculate which questions should show/still need to be answered in the Check Answers page, and it's used on the Tasklist page to show the status of each section.

I'd agree that having the form know about the case log here is possibly a slightly leaky abstraction but I think this is the most logical place for it to sit for now? Open to suggestion though.

Reply:
I haven't quite figured out how this could be done differently, so happy for this to be there till a time when a logical refactor comes our way

Reply:
So happy == therefore happy and not == very happy